Source/Immutable/Public/Immutable/ApplicationConfig.h

```diff
@@ -0,0 +1,162 @@
+﻿#pragma once
+
+#include "ApplicationConfig.generated.h"
+
+/**
+ * @class UApplicationConfig
+ * @brief Configuration settings for Passport and various APIs.
+ * @details This class stores configuration settings such as URLs, chain names, contract addresses, 
+ * client IDs, and environment settings for the zkEVM API, Orderbook API, and Passport.
+ */
+UCLASS(Abstract, Blueprintable, ClassGroup = Immutable)
+class UApplicationConfig : public UObject
```

Comment:
I'm thinking, can't you split this up into at least 2-3 different structs ?

```cpp
/** Passport Client ID. */
UPROPERTY(EditDefaultsOnly, Category = "Passport")
FString ClientID;

/** Environment used to initialize passport. Ex. sandbox or production */
UPROPERTY(EditDefaultsOnly, Category = "Passport")
FString Environment;

/**
* (Android, iOS, and macOS only)
* The URL where the browser will redirect after successful authentication.
*/
UPROPERTY(EditDefaultsOnly, Category = "Passport")
FString RedirectURL;

/** The URL where the browser will redirect after logout is complete. */
UPROPERTY(EditDefaultsOnly, Category = "Passport")
FString LogoutURL;
```
For example, one for passport
And you can use this generally and everywhere for passport related things e.g. initialising passport

Reply:
My argument would be to keep everything in a single place. I see some benefits in terms of readability, but having a single BP config asset rather than multiple outweighs the readability advantage.